sources/notation-image-verifier/main.go

```diff
@@ -0,0 +1,46 @@
+// package main provides a simple wrapper of the notation CLI adhering to the
+// containerd image verification plugin interface.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	var digest = flag.String("digest", "", "image digest to verify")
+	var name = flag.String("name", "", "image name to verify")
+	var _ = flag.String("stdin-media-type", "", "image media type")
+	flag.Parse()
+
+	if *digest == "" || *name == "" {
+		// containerd only captures the stdout of image verification binaries, hence no
+		// write to stderr here (and elsewhere in the program).
+		fmt.Printf("Usage: %s -digest <digest> -name <name>\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	fmt.Printf("verifying image: %s\n", *name)
+
+	// Override the default notation paths to what's packaged in packages/notation
+	os.Setenv("NOTATION_CONFIG", "/etc/notation")
+	os.Setenv("NOTATION_CACHE", "/var/cache/notation")
+	os.Setenv("NOTATION_LIBEXEC", "/usr/bin/notation-plugins")
+
+	// The notation CLI looks in $HOME/.docker/config.json for credentials.
+	// Bottlerocket doesn't set a $HOME.
+	os.Setenv("HOME", "/root")
+
+	cmd := exec.Command("notation", "verify", *name)
```

Comment:
Is there a problem if you use the full path to `notation`? At least that's what we do in rust programs when we call other helper binaries.

Reply:
good call, I'm going to update to the absolute path

I'll keep this unresolved as I keep getting through the lower hanging fruit feedback - I'll have to consider the FIPS case for that path.

Reply:
NVM - I'll point to `/usr/bin/notation`, we have an overlay that already handles this :D https://github.com/bottlerocket-os/bottlerocket-core-kit/blob/39a939781bd126aeaf3ef7e3b3185f6c18a8b439/packages/release/usr-bin.mount.in#L11